File: data/code_pairs/fortran/2447.f90
module mytype_mod
  implicit none
  type :: mytype
    integer, allocatable :: indices(:)
  end type mytype
end module mytype_mod

program main
  use mytype_mod
  implicit none

  type(mytype) :: object
  integer :: i, me

  ! For a standalone example, simulate 'this_image()' with a fixed value
  me = 1

  ! Allocate and set indices array
  allocate(object%indices(me))
  object%indices = [(i, i = 1, me)]

  ! Perform checks
  if (size(object%indices) /= 1) then
    call abort()
  endif
  if (object%indices(1) /= 1) then
    call abort()
  endif

  print *, "Fortran Unit Test Passed Successfully."
end program main

File: data/code_pairs/cpp/2447.cpp
#include <iostream>
#include <vector>

struct MyType {
    std::vector<int> indices;
};

int main() {
    MyType object;

    // Simulating a single "process" scenario, akin to Fortran's 'this_image()' for standalone examples.
    int me = 1; // Normally, this would be dynamically determined in a parallel environment.

    // Fill the vector.
    for (int i = 1; i <= me; ++i) {
        object.indices.push_back(i);
    }

    // Perform checks.
    if (object.indices.size() != 1) {
        std::cerr << "Error: Size of indices is not 1." << std::endl;
        return 1; // Error exit.
    }

    if (object.indices[0] != 1) {
        std::cerr << "Error: First element of indices is not 1." << std::endl;
        return 1; // Error exit.
    }

    std::cout << "C++ Unit Test Passed Successfully." << std::endl;
    return 0; // Success exit.
}